From ca60f762c8a84bfe406c58494dbf8c2b65adb1bc Mon Sep 17 00:00:00 2001
From: Zan Dobersek <zdobersek@igalia.com>
Date: Fri, 2 Jan 2026 13:28:23 +0100
Subject: [PATCH] tu: enable storageBuffer8BitAccess on all a7xx hardware

Move the enabled storage_8bit property toggle into the base a7xx GPUProps
class. This enables storageBuffer8BitAccess Vulkan feature on all a7xx
hardware, much like the proprietary driver does. It's also a required
feature with Vulkan 1.4.

Fixes: dEQP-VK.info.device_mandatory_features on pre-a750 a7xx hardware.

Signed-off-by: Zan Dobersek <zdobersek@igalia.com>
---
 src/freedreno/common/freedreno_devices.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/freedreno/common/freedreno_devices.py b/src/freedreno/common/freedreno_devices.py
index a2637b0815cc4..55c4077f3abf4 100644
--- a/src/freedreno/common/freedreno_devices.py
+++ b/src/freedreno/common/freedreno_devices.py
@@ -958,6 +958,7 @@ a7xx_base = GPUProps(
         has_z24uint_s8uint = True,
         tess_use_shared = True,
         storage_16bit = True,
+        storage_8bit = True,
         has_tex_filter_cubic = True,
         has_separate_chroma_filter = True,
         has_sample_locations = True,
@@ -1026,7 +1027,6 @@ a7xx_gen3 = GPUProps(
         has_generic_clear = True,
         r8g8_faulty_fast_clear_quirk = True,
         gs_vpc_adjacency_quirk = True,
-        storage_8bit = True,
         ubwc_all_formats_compatible = True,
         has_compliant_dp4acc = True,
         ubwc_coherency_quirk = True,
-- 
GitLab

